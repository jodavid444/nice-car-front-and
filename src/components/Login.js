import { Link } from 'react-router-dom';
import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { loginUser } from '../redux/users/usersSlice';
import './LoginForm.css';

export const LoginForm = () => {
  const dispatch = useDispatch();
  const [username, setUsername] = useState('');

  const handleLogin = (e) => {
    e.preventDefault();
    dispatch(loginUser({ name: username }));
  };

  return (
    <div className="login-container">
      <div className="login-form">
        <h2 className="login-heading">Login</h2>
        <form onSubmit={handleLogin}>
          <div className="form-group">
            <input
              type="text"
              className="form-control"
              placeholder="Username"
              value={username}
              onChange={(e) => setUsername(e.target.value)}
            />
          </div>
          <button type="submit" className="btn btn-primary btn-login">Login</button>
        </form>
        <h2 className="or-heading">Or</h2>
        <Link to="/registration" className="link">
          <button type="button" className="btn btn-secondary btn-register">Register</button>
        </Link>
      </div>
    </div>
  );
};

export default LoginForm;
