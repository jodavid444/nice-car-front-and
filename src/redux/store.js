import { configureStore } from '@reduxjs/toolkit';
import carsReducer from './cars/carsSlice';
import reservationsReducer from './reservations/reservationsSlice';
import usersReducer from './users/usersSlice';

const store = configureStore({
  reducer: {
    cars: carsReducer,
    reservations: reservationsReducer,
    users: usersReducer,
  },
});
export default store;
