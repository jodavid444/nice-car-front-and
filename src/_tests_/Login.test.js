import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import Login from '../components/Login';
import store from '../redux/store';

describe('LoginForm component', () => {
  it('should match the snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <Router>
          <Login />
        </Router>
      </Provider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
