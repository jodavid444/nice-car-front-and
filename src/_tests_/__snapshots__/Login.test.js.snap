// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component should match the snapshot 1`] = `
<div>
  <h2>
    Login
  </h2>
  <form>
    <input
      placeholder="Username"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Login
    </button>
  </form>
  <h2>
    Or
  </h2>
  <a
    class="link"
    href="/registration"
  >
    <button
      type="button"
    >
      Register
    </button>
  </a>
</div>
`;
